redbot/cogs/alias/alias.py

```diff
@@ -292,17 +293,43 @@ async def _list_alias(self, ctx: commands.Context):
         guild_aliases = await self._aliases.get_guild_aliases(ctx.guild)
         if not guild_aliases:
             return await ctx.send(_("There are no aliases on this server."))
-        names = [_("Aliases:")] + sorted(["+ " + a.name for a in guild_aliases])
-        await ctx.send(box("\n".join(names), "diff"))
+        names = [""] + sorted(["+ " + a.name for a in guild_aliases])
+        message = ""
+        for a in names:
+            message += "{}\n".format(a)
+        alias_list = []
+        for page in pagify(message, delims=["\n"], page_length=1900):
+            page = (_("Aliases:")) + page
+            alias_list.append(box("".join(page), "diff"))
+        if len(alias_list) == 1:
+            return await ctx.send(alias_list[0])
+        if ctx.guild.me.permissions_in(ctx.channel).add_reactions:
+            await menu(ctx, alias_list, DEFAULT_CONTROLS)
+        else:
+            for i in range(len(alias_list)):
+                await ctx.send("{}".format(alias_list[i]))
 
     @global_.command(name="list")
     async def _list_global_alias(self, ctx: commands.Context):
         """List the available global aliases on this bot."""
         global_aliases = await self._aliases.get_global_aliases()
         if not global_aliases:
             return await ctx.send(_("There are no global aliases."))
-        names = [_("Aliases:")] + sorted(["+ " + a.name for a in global_aliases])
-        await ctx.send(box("\n".join(names), "diff"))
+        names = [""] + sorted(["+ " + a.name for a in global_aliases])
+        message = ""
+        for a in names:
+            message += "{}\n".format(a)
+        alias_list = []
+        for page in pagify(message, delims=["\n"], page_length=1900):
+            page = (_("Aliases:")) + page
+            alias_list.append(box("".join(page), "diff"))
+        if len(alias_list) == 1:
+            return await ctx.send(alias_list[0])
+        if ctx.guild.me.permissions_in(ctx.channel).add_reactions:
+            await menu(ctx, alias_list, DEFAULT_CONTROLS)
+        else:
+            for i in range(len(alias_list)):
+                await ctx.send("{}".format(alias_list[i]))
```

Comment:
IMO, we should just not allow the command to run if bot doesn't have `add_reactions` perm (using `@commands.bot_has_permissions` deco) but if you think it *should* allow that, you should use `ctx.send_interactive()` to avoid a lot of message spam.

```suggestion
        else:
            await ctx.send_interactive(alias_list)
```